From e71b68c04ed7b5e0a6cc983600812785e947bb56 Mon Sep 17 00:00:00 2001
From: Yin Fengwei <fengwei.yin@intel.com>
Date: Thu, 26 Aug 2021 12:31:06 +0800
Subject: [PATCH] crashlog: fix build issue with e2fsprogs v1.46.2

In e2fsprogs v1.46.2, s_volume_name is defined with attribute
nonstring. So strncmp instead of strcmp should be used to do
comparing.

Upstream-Status: Pending

Signed-off-by: Yin Fengwei <fengwei.yin@intel.com>
Signed-off-by: Naveen Saini <naveen.kumar.saini@intel.com>
---
 misc/debug_tools/acrn_crashlog/acrnprobe/loop.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/misc/debug_tools/acrn_crashlog/acrnprobe/loop.c b/misc/debug_tools/acrn_crashlog/acrnprobe/loop.c
index ad6db22ba..ee27e4aa6 100644
--- a/misc/debug_tools/acrn_crashlog/acrnprobe/loop.c
+++ b/misc/debug_tools/acrn_crashlog/acrnprobe/loop.c
@@ -231,7 +231,7 @@ int loopdev_check_parname(const char *loopdev, const char *parname)
 		/* only look into the primary super block */
 		if (super.s_volume_name[0]) {
 			close(fd);
-			return !strcmp((const char *)super.s_volume_name, parname);
+			return !strncmp((const char *)super.s_volume_name, parname, EXT2_LABEL_LEN);
 		}
 		break;
 	}
-- 
2.17.1

